ios/Classes/components/applePay/BaseApplePayComponent.swift

```diff
@@ -0,0 +1,19 @@
+@_spi(AdyenInternal)
+import Adyen
+
+class BaseApplePayComponent {
+    var applePayComponent: ApplePayComponent?
+
+    func present() {
+        preconditionFailure("This method must be overridden")
+    }
+    
+    func getViewController() -> UIViewController? {
+        var rootViewController = UIApplication.shared.adyen.mainKeyWindow?.rootViewController
+        while let presentedViewController = rootViewController?.presentedViewController {
```

Comment:
Check out the `UIViewControllerHelpers` in the Adyen module and use that code here